server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java

```diff
@@ -608,9 +604,11 @@ private PhysicalPlan transformQuery(QueryOperator queryOperator, int fetchSize)
         }
       }
     }
-    if(queryOperator instanceof QueryIndexOperator) {
-      ((QueryIndexPlan) queryPlan).setIndexType(((QueryIndexOperator) queryOperator).getIndexType());
-      ((QueryIndexPlan) queryPlan).setProps(((QueryIndexOperator) queryOperator).getProps());
+    if(queryOperator.getIndexType() != null) {
+      if(queryPlan instanceof QueryIndexPlan) {
```

Comment:
As initialized at line 397, will `getIndexType() != null` and `instanceof QueryIdnexPlan` be equivalent?

Reply:
I think it's equivalent